### ISSUE TITLE
Unable to delete TargetGroupBinding if Target Group is missing

### ISSUE THREAD

**Issue by tmehlinger** [kubernetes-sigs/aws-load-balancer-controller#2341]
**Describe the bug**
I'm receiving the following errors in the ALB controller after creating a TGB resource with an incorrect Target Group ARN:

```
{"level":"error","ts":1636048222.6456954,"logger":"controller-runtime.manager.controller.targetGroupBinding","msg":"Reconciler error","reconciler group":"elbv2.k8s.aws","reconciler kind":"TargetGroupBinding","name":"websocket-ws-3ovam125","namespace":"assistant","error":"ValidationError: 'arn:aws:elasticloadbalancing:us-east-2:159818664873:targetgroup/websocketWsTg-183db7a/4e446602feb6d777' is not a valid target group ARN\n\tstatus code: 400, request id: 7aeb81da-583c-413f-833b-1c29a5a75ddd"}
```

When I try to delete the offending TGB, it repeats the above error and never deletes the resource.

**Steps to reproduce**

Create a `TargetGroupBinding` resource with an syntactically-valid but nonexistent Target Group ARN.

**Expected outcome**

Log a warning that the TG ARN doesn't exist, then delete the TGB resource.

**Environment**

* AWS Load Balancer controller version: **2.2.0 and 2.3.0**
* Kubernetes version: **1.19**
* Using EKS (yes/no), if so version? **eks.6**

**Comment by tmehlinger** [comment]
I was actually able to work around this following the advice here: https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/1629#issuecomment-731011683

The controller wouldn't delete the resources because it couldn't finalize them, so editing the TGB resources to remove the finalizers allowed the API server to clean them up.

**Comment by M00nF1sh** [comment]
/kind bug
It looks like you specified an invalid TargetGroup ARN instead of not found one.
Currently we only handled not found TargetGroup case: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/c60d5e63485fe1c4dd726c5aa0fff8bb6459e073/pkg/targetgroupbinding/resource_manager.go#L203

we should add a check for invalid TargetGroup ARN error as well and ignore it.

**Comment by absmith82** [comment]
I'm also having this issue and removing the finalizers does delete the ingress, but the ingress controller still tries to find the target group even though it doesn't exist.

**Comment by absmith82** [comment]
was able to resolve my issue, delete the targetgroupbinding resource in kubernetes, and there was another ingress that was causing issues rebuilding the alb ingress controller configuration, once I removed that ingress everything started working fine again.

**Comment by kishorj** [comment]
Fix is available in v2.4.0 release, closing the issue.

**Comment by omidraha** [comment]
Same issue described here: https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/1629#issuecomment-1620822133
